invoice/src/main/java/org/killbill/billing/invoice/generator/InvoiceWithMetadata.java

```diff
@@ -155,6 +154,16 @@ public boolean apply(final InvoiceItem input) {
         });
     }
 
+    private boolean hasItemsForDate(final LocalDate date) {
```

Comment:
Don't really understand what this is about, and why this was missing 🤔

Reply:
When only a `REPAIR_ADJ` is generated, we were clearing the notifications entirely. As a side effect, we were not generating the invoice at the next BCD (2016-7-17) in `testAccountBCDChangeWithNoOptimization` -- we were waiting until the 20th.